### PR TITLE
Merge headers of ocs results

### PR DESCRIPTION
**Pull request by DeepDiver1975**
@rullzer as requested

### CODE REVIEW

File: lib/private/ocs/result.php [hunk 1]
```diff
@@ -51,7 +51,7 @@ class OC_OCS_Result{
 	 * @param int $code
 	 * @param null|string $message
 	 */
-	public function __construct($data=null, $code=100, $message=null) {
+	public function __construct($data = null, $code = 100, $message = null, $headers = []) {
```
Comment (LukasReschke) [inline lib/private/ocs/result.php]: PHPDoc also `array $headers`?

**Comment by rullzer** [comment]
Works!

**Comment by rullzer** [comment]
So this could potentially break stuff. Since assume we have 2 routes both to ocs/v1.php/foo and the first set a header ```FOO: BAR``` and the second a header ```FOO: FOOBAR``` then the header will be the one that is merged the latests.

Which could be all wrong if the header sets something about the returned data (like the pagination stuff).

In other words is the mering of responses used anywhere else except in the capabilities (which are no longer needed due to the capabilities manager)?

**Comment by DeepDiver1975** [comment]
> In other words is the mering of responses used anywhere else except in the capabilities (which are no longer needed due to the capabilities manager)?

afaik not

**Comment by DeepDiver1975** [comment]
some more reviewers pleas @PVince81 @LukasReschke @blizzz THX

**Comment by PVince81** [comment]
@nickvergessen

**Comment by DeepDiver1975** [comment]
@nickvergessen @PVince81 @rullzer merge or not?

**Comment by nickvergessen** [comment]
We don't need it atm? Or does it solve a bug?

**Comment by rullzer** [comment]
It is just to make sure that the headers we set in the sharee API are properly propogated. 9.0 is fine by me.

**Comment by PVince81** [comment]
:+1:

**Pull request merged**